Neos.ContentGraph.DoctrineDbalAdapter/src/DoctrineDbalContentGraphProjection.php

```diff
@@ -259,6 +259,19 @@ public function apply(EventInterface $event, EventEnvelope $eventEnvelope): void
         }
     }
 
+    public function inSimulation(\Closure $fn): mixed
+    {
+        if ($this->dbal->isTransactionActive()) {
```

Comment:
did you try to get the `setRollbackOnly()` working (potentially on a clone of the $dbal instance)?
Otherwise I would fear that some code somewhere commits the transaction

Reply:
i mean we _could_ set it to true and then use reflection to unset it again 😅 
but i dont see a way how we can use a clone here ... except if we were to set the clone on $this->dbal = $clone ... but that would also be ugly?

Reply:
maybe we can wrap the instance in a decorator on creation of the whole projection?

Reply:
we decided in a discussion to leave it be for the moment and come back to how we want to split / protect the connectoin at a later point.